drivers/input/input_cap1203.c

```diff
@@ -96,7 +95,13 @@ static int cap1203_process(const struct device *dev)
 		return r;
 	}
 
-	data->callback(dev, 0, col, pressed);
+	if (pressed) {
+		input_report_abs(dev, INPUT_ABS_X, col, false, K_FOREVER);
+		input_report_abs(dev, INPUT_ABS_Y, 0, false, K_FOREVER);
+		input_report_key(dev, INPUT_BTN_TOUCH, 1, true, K_FOREVER);
+	} else {
+		input_report_key(dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
+	}
```

Comment:
How would you feel about reworking this one to generate `input_report_key` events instead? This specific driver was somewhat fitting poorly in the keyscan API, so I think that we should convert it properly.

The change would be:
- add a devicetree property to take three input codes and store it in the config structure (you can look at the longpress code for that)
- add a field in the data structure to store the old input state
- in the callback code here compare the new `input` state with the old one, and for every bit that change fire up a `input_report_key` event for the corresponding code and with the right value depending on the new state

That will make it behave the same as `gpio-keys`, down the road we can plumb that in a `LV_INDEV_TYPE_BUTTON` thing.

Reply:
I noticed that it was not a kscan driver in the "traditional" sense. If you are ok with breaking downstream, I will perform your suggested changes :^). I'd first commit this separately and if the results are ok, squash it down.

Reply:
Yeah, I'll check with the architecture folks but I think it's the right thing to do for this.

Fine to do it both in one or two commits, up to you really. Thanks!!